lib/huginn_acumen_product_agent/concerns/product_contributors_query_concern.rb

```diff
@@ -0,0 +1,118 @@
+    contributor_type_data = process_contributor_type_response(contributor_type_data)
+
+    return map_contributor_data(products, contributor_data, contributor_type_data)
+  end
+
+  # This function parses the raw data returned from the ProdMkt_Contrib_Link table
+  # This table holds the relationship between products and Contributors
+  # The resulting data is a hash mapping contributor arrays to Prod_Mkt.ID values
+  def process_product_contributor_response(raw_data)
+    contributors = []
+
+    raw_data.each do |contributor|
+
+      begin
+        mapped = response_mapper(contributor, {
+          'ProdMkt_Contrib_Link.ProdMkt_Contrib_ID' => 'contributor_id',
+          'ProdMkt_Contrib_Link.ProdMkt_ID' => 'product_marketing_id',
+          'ProdMkt_Contrib_Link.Inactive' => 'inactive',
+        })
+
+        if mapped['inactive'] == '0'
+          contributors.push(mapped)
+        end
+      rescue => error
+        issue_error(AcumenAgentError.new(
+          'process_product_contributor_response',
+          'Failed while processing contributor record',
+          contributor,
+          error,
+        ))
+      end
+    end
+
+    return contributors
+  end
+
+  # This function parses the raw data returned from the ProdMkt_Contributor table
+  # This table holds the contributor type (e.g. Author) for the
+  # contributor/product relationship
+  def process_contributor_type_response(raw_data)
+    results = {}
+    raw_data.map do |contributor_type|
+
+      begin
+        mapped = response_mapper(contributor_type, {
+          # It's important to note that this ID is the record on the _relationship_
+          # table. It is _not_ the actual Contributors ID.
+          'ProdMkt_Contributor.ID' => 'contributor_relation_id',
+          'ProdMkt_Contributor.Contrib_Type' => 'type',
+        })
+
+
+        if !results[mapped['contributor_relation_id']]
+          results[mapped['contributor_relation_id']] == mapped['type']
+        end
+      rescue => error
+        issue_error(AcumenAgentError.new(
+          'process_contributor_type_response',
+          'Failed while processing contributor type record',
+          contributor_type,
+          error,
+        ))
+      end
+    end
+
+    return results
+  end
+
+  # This function maps parsed Contributor records to their matching Inv_Product record
+  def map_contributor_data(products, contributor_data, contributor_type_data)
+    products.each do |product|
+
+      begin
+        marketing_id = product['acumenAttributes']['product_marketing_id']
+        contributors = contributor_data.select { |c| c['product_marketing_id'] == marketing_id }
+
+        product['contributors'] = contributors.map do |c|
+          {
+            '@type' => 'Person',
+            'identifier' => c['contributor_id'],
+            'acumenAttributes' => {
+              'contrib_type' => contributor_type_data[c['contributor_id']]
```

Comment:
@mrogers-5s This is not working, I am seeing null for products that this field should not be null on. Specifically look at the product, 12824